### PR TITLE
Verify docker.download.com is reachable before attempting to install the Docker Package Repo in 1-install-dependencies.sh

### PR DESCRIPTION
**Pull request by raul-marquez-csa**
The network randomly fails after the service restarts that happen before installing the **Docker Package Repo**, this ultimately makes the TH installation fail.

A ping to `docker.download.com` will be attempted and retried in increments of 1 second before a 5 minute timeout.

### CODE REVIEW

File: scripts/ubuntu/1-install-dependendcies.sh [hunk 1]
```diff
@@ -64,4 +96,9 @@ done
 IFS=$SAVEIFS 
 
 # Install Peotry, needed for Test Harness CLI
+echo
+echo "####################################"
+echo "Install Peotry, needed for Test Harness CLI"
```
Comment (hiltonlima) [inline scripts/ubuntu/1-install-dependendcies.sh:101]:
```suggestion
echo "Install Poetry, needed for Test Harness CLI"
```

File: scripts/ubuntu/1-install-dependendcies.sh [hunk 2]
```diff
@@ -17,10 +17,42 @@
...
+    echo "Ping docker.download.com successful"
+    end_time=$(date)
+    echo "Ping ended at: $end_time"
+    echo "Ping duration: $(($(date +%s) - $(date -d "$start_time" +%s))) seconds"
+    break
+  fi
+  echo "Ping docker.download.com failed, retrying..."
+  sleep 1
+done
+if [ $? -eq 124 ]; then
```
Comment (hiltonlima) [inline scripts/ubuntu/1-install-dependendcies.sh:39]: Hi @raul-marquez-csa ,

Could you explain why you are testing for this specific return value?

**Review by hiltonlima — COMMENTED** [review]
We will soon release the new version of TH.
This version is in the `fall2024` branch.

There we have already made some changes to the scripts that you are suggesting to change in this PR. Could you change the target of this PR to the `fall2024` branch?

File: scripts/ubuntu/1-install-dependendcies.sh [hunk 3]
```diff
@@ -17,10 +17,42 @@
...
+  end_time=$(date)
+  echo "docker.download.com: Timeout reached"
+  echo "Ping ended at: $end_time"
+  echo "Ping duration: $(($(date +%s) - $(date -d "$start_time" +%s))) seconds"
+fi
+'
+
+curl -fsSL --retry 2 --retry-connrefused -v https://download.docker.com/linux/ubuntu/gpg | sudo gpg --dearmor --yes -o /usr/share/keyrings/docker-archive-keyring.gpg
```
Comment (antonio-amjr) [inline scripts/ubuntu/1-install-dependendcies.sh:47]: Maybe the PR description is outdated, but then we don't need to use the `-4fsSL` anymore?
The difference here is the retry options plus the ping test before?

File: scripts/ubuntu/1-install-dependendcies.sh (hunk 1 above)
Comment (raul-marquez-csa) [reply scripts/ubuntu/1-install-dependendcies.sh]: This has been fixed in a separate commit

File: scripts/ubuntu/1-install-dependendcies.sh (hunk 2 above)
Comment (raul-marquez-csa) [reply scripts/ubuntu/1-install-dependendcies.sh:37]: 124 is the timeout reached code

File: scripts/ubuntu/1-install-dependendcies.sh (hunk 3 above)
Comment (raul-marquez-csa) [reply scripts/ubuntu/1-install-dependendcies.sh]: This line has been restored to original

File: scripts/ubuntu/1-install-dependendcies.sh (hunk 2 above)
Comment (raul-marquez-csa) [reply scripts/ubuntu/1-install-dependendcies.sh:37]: Is it possible to still merge this to the spring branch? There's an upcoming rapid recert training in June that uses the sprint release and would help not having this issue as there will be many members to support.

**Review by hiltonlima — APPROVED** [review]

**Review by rquidute — APPROVED** [review]

**Review by antonio-amjr — APPROVED** [review]

**Comment by hiltonlima** [comment]
All good :) 
Fell free to squash and merge it.

@raul-marquez-csa

File: scripts/ubuntu/1-install-dependendcies.sh [hunk 4]
```diff
@@ -16,9 +16,34 @@
...
+  echo "docker.download.com: Timeout reached"
+  echo "Ping ended at: $end_time"
+  echo "Ping duration: $(($(date +%s) - $(date -d "$start_time" +%s))) seconds"
+fi
+'
+
 # Install Docker Package Repo
-curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo gpg --dearmor --yes -o /usr/share/keyrings/docker-archive-keyring.gpg
-echo "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/docker-archive-keyring.gpg] https://download.docker.com/linux/ubuntu $(lsb_release -cs) stable" | sudo tee /etc/apt/sources.list.d/docker.list > /dev/null
+curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo gpg --dearmor --yes -o /usr/share/keyrings/docker-archive-keyring.gpgecho "deb [arch=$(dpkg --print-architecture) signed-by=/usr/share/keyrings/docker-archive-keyring.gpg] https://download.docker.com/linux/ubuntu $(lsb_release -cs) stable" | sudo tee /etc/apt/sources.list.d/docker.list > /dev/null
```
Comment (JuliaPageMorseMicro) [inline scripts/ubuntu/1-install-dependendcies.sh:46]: The lost line feed between these two lines caused the script to fail.

**Pull request merged**